### PR TITLE
Complete unit test coverage for `Bundler::Fetcher::Index`

### PR DESCRIPTION
**Pull request by RochesterinNYC**

### CODE REVIEW

File: spec/bundler/fetcher/index_spec.rb [hunk 1]
```diff
@@ -1,16 +1,97 @@
...
+    subject.specs(gem_names)
+  end
+
+  context "error handling" do
+    shared_examples_for "the error is properly handled" do
+      context "when certificate verify failed" do
+        let(:error_message) { "certificate verify failed" }
+
+        it "should raise a Bundler::Fetcher::CertificateFailureError" do
+          expect { subject.specs(gem_names) }.to raise_error(Bundler::Fetcher::CertificateFailureError, %r{Could not verify the SSL certificate for http://sample_uri.com})
```
Comment (segiddins) [inline spec/bundler/fetcher/index_spec.rb]: break this onto 2 lines

File: spec/bundler/fetcher/index_spec.rb [hunk 2]
```diff
@@ -1,16 +1,97 @@
...
+
+        it "should raise a Bundler::Fetcher::AuthenticationRequiredError" do
+          expect { subject.specs(gem_names) }.to raise_error(Bundler::Fetcher::AuthenticationRequiredError, %r{Authentication is required for http://remote_uri.org})
+        end
+      end
+
+      context "when a 403 response occurs" do
+        let(:error_message)     { "403" }
+        let(:remote_uri)        { double(:remote_uri) }
+        let(:remote_uri_string) { "http:\/\/remote_uri\.org" }
```
Comment (segiddins) [inline spec/bundler/fetcher/index_spec.rb]: use single quoted strings?

**Comment by segiddins** [comment]
:+1:

**Comment by segiddins** [comment]
@homu r+

**Comment by RochesterinNYC** [comment]
Quick ?, does homu run the full test suite in Travis CI and if so, do we have access to see it as it's running?

**Comment by segiddins** [comment]
It will run it on travis, yeah

**Pull request merged**